packages/amesos2/src/Amesos2_Tacho_def.hpp

```diff
@@ -208,21 +208,11 @@ TachoSolver<Matrix,Vector>::solve_impl(const Teuchos::Ptr<MultiVecAdapter<Vector
     int i_ld_rhs = as<int>(ld_rhs);
 
     for(size_t j = 0 ; j < nrhs; j++) {
-      // TODO: Decide how to organize this best for Amesos2
-      // Clarify behavior for Cuda - exception? configure error?
-#ifdef KOKKOS_HAVE_OPENMP
-      typedef Kokkos::OpenMP                                  DeviceSpaceType;
-#else
-      typedef Kokkos::Serial                                  DeviceSpaceType;
-#endif
-      typedef typename Tacho::Solver<tacho_type,DeviceSpaceType>::
-        value_type_matrix solve_array_t;
-
       const int n = 1;
       solve_array_t x(&xValues.getRawPtr()[j*i_ld_rhs], this->globalNumRows_, n);
       solve_array_t b(&bValues.getRawPtr()[j*i_ld_rhs], this->globalNumRows_, n);
-      solve_array_t t("t", this->globalNumRows_, n);
-      data_.solver.solve(x, b, t);
+
+      data_.solver.solve(x, b, workspace_);
 
```

Comment:
So, Do we always assume that we have a single rhs ? Is Tpetra disable to use multiple rhs ? If nrhs is not 1, I would do 
```
const int nrhs = from_tpetra_multivector_num_columns;
if (t.dimension_0() < this->globalNumRows() || t.dimension_1() < nrhs) {
  t = solve_array_t("t", this->globalNumRows, nrhs);
}
data_.solver.solve(x, b, t);
```

Reply:
I will fix this so the solver is not passing them through one at a time. I'll build the workspace as you suggested so it only resizes when necessary.

Reply:
@kyungjoo-kim Fixed so it passes the full set of vectors to Tacho now.